aiosmtpd/tests/test_smtp.py

```diff
@@ -434,6 +434,13 @@ def test_mail_params_bad_syntax_esmtp(self):
                 response,
                 b'Syntax: MAIL FROM: <address> [SP <mail-parameters>]')
 
+    def test_mail_fail_parse_email(self):
+        with SMTP(*self.address) as client:
+            client.helo('example.com')
+            code, response = client.docmd('MAIL FROM: <""@example.com>')
```

Comment:
I wasn`t thinking about it, as it is not changed too long. There is no example that must fail with index error, as that is not expected to be. If it will be fixed, there will be no example to cover IndexError. May be I should mock something to guarantee raise of exception.